Actions.js

```diff
@@ -21,64 +21,86 @@ router.get('/:id', (req, res) => {
     db.get( id )
 
      .then(action => {
+       if (!id) 
+       res.status(404).json({
+         message: "Could not find action with this ID."
+       })
+       else {
             res.status(200).json(action)
           }
-     )
+     })
     .catch(err => {
         res.status(500).json({
-          message: 'Could not retrieve action.',
+          message: 'Server error- Could not retrieve action.',
           err
         })
     })
   })
 
   
-router.post('/', (req, res) => {
-    const { object } = req.body;
-    db.insert({object})
+router.post('/:id', (req, res) => {
+    const newAction, { project_id, description, notes } = req.body;
+
+    console.log(newAction);
+
+    db.insert(newAction)
   
     .then(added => {
+      if (!project_id || !description || !notes) {
+        res.status(404).json({
+          message: "Please include project_id, description, and notes"
+        })
+      } else {
```

Comment:
the ordering of error checks is not correct.  You are inserting the action, then you are validating the action that was sent from the body after the insert has finished.   You need to validate the action first.  After the req.body has been verified, it can then be inserted